helix-core/src/main/java/org/apache/helix/controller/dataproviders/ResourceControllerDataProvider.java

```diff
@@ -573,11 +575,16 @@ public WagedInstanceCapacity getWagedInstanceCapacity() {
     return _wagedInstanceCapacity;
   }
 
-  private void buildSimpleCapacityMap(int globalMaxPartitionAllowedPerInstance) {
+  private void buildSimpleCapacityMap() {
+    ClusterConfig clusterConfig = getClusterConfig();
+    ClusterTopologyConfig clusterTopologyConfig =
+        ClusterTopologyConfig.createFromClusterConfig(clusterConfig);
+    Map<String, InstanceConfig> instanceConfigMap = getAssignableInstanceConfigMap();
     _simpleCapacitySet = new HashSet<>();
-    for (String instance : getEnabledLiveInstances()) {
-      CapacityNode capacityNode = new CapacityNode(instance);
-      capacityNode.setCapacity(globalMaxPartitionAllowedPerInstance);
+    for (String instanceName : getEnabledLiveInstances()) {
```

Comment:
Why don't disabled and down instances get included in the capacitySet? This will cause all the replicas on the down or disabled nodes to be reassigned somewhere else in the cluster if delayed rebalance is being used. We should have all assignable instances in the map and leave it to the rebalancer to pass down liveNodes to `computePartitionAssignment`.

Reply:
> This will cause all the replicas on the down or disabled nodes to be reassigned somewhere else in the cluster if delayed rebalance is being used

If the node is disabled or down, do we expect to re-assign the replica to other nodes?

Reply:
This behavior is determined by the rebalancer and not rebalance strategy. If you use DelayedAutoRebalancer it should not try to reassign the replicas to other nodes until after the delay window. If you use AutoRebalancer, it should.

If we only have EnabledLive in capacitySet, we will make the behavior as I described for AutoRebalancer even when DelayedAutoRebalancer is used. This will be inconsistent behavior with assumption of how each rebalancer should work.

Reply:
Got you, thanks for the detailed context, will make the changes